openedx_ledger/models.py

```diff
@@ -185,6 +187,28 @@ def balance(self, committed_only=False):
             queryset = Transaction.objects.filter(ledger=self).exclude(state=TransactionStateChoices.FAILED)
         return self.subset_balance(queryset)
 
+    def total_deposits(self):
+        """
+        Calculate and return the total value added to this ledger.
+
+        Note that this not only factors actual deposits, but also all adjustments which essentially represent
+        ad-hoc manipulations of the total value added.
+        """
+        # Custom filter that searches for anything that looks and smells like a deposit, including actual deposits.
+        deposit_esque_filter = Q(deposit__isnull=False) | Q(adjustment__isnull=False)
+        # Also make sure to include the initial deposit since there will be a transition period during which initial
+        # deposits won't yet have an actual related Deposit record.
+        # We can remove this after backfilling or introducing a data migration (ENT-9132).
+        deposit_esque_filter |= Q(idempotency_key__endswith=INITIAL_DEPOSIT_TRANSACTION_SLUG)
```

Comment:
+1 on the removal of this as it will make the query pretty slow.

Reply:
hm, so an endswith filter on an indexed char field takes a lot of time?

Reply:
Interesting! Char fields are indexed by MySQL left-to-right, so it seems that checking for something at the end of a string might be inefficient. Especially in our case the characters before the slug are extremely high cardinality (hash digest), so it effectively has no clue as to which index branch to take.

Reply:
so I can think of 2 options:
1. Stay the course and hope performance doesn't tank. Followup fast with a data migration PR and code cleanup PR.
2. Change the order of implementation such that all business logic stays the same, but we add the Deposit model and backfill initial transactions. Then followup later with the remainder of the business logic changes.

Reply:
going with option 2. Deleted this line of code.